src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -0,0 +1,95 @@
+package nextstep.subway.section.domain;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import javax.persistence.Transient;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import nextstep.subway.common.exception.ExistException;
+import nextstep.subway.common.exception.NothingException;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.station.domain.Station;
+
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@Getter
+@Embeddable
+public class Sections {
+
+	@OneToMany(mappedBy = "line", cascade = CascadeType.ALL) //, orphanRemoval = true
+	private final List<Section> sections = new ArrayList<>();
+
+	@Transient
+	private List<Station> stations;
+
+	public Sections(Section section) {
+		this.sections.add(section);
+	}
+
+	public static Sections of(Line line, Station upStation, Station downStation, int distance) {
+		return new Sections(new Section(line, upStation, downStation, distance));
+	}
+
+	public void add(Section target) {
+		validateSection(target);
+
+		if (isExistUpStation(target)) {
+			this.sections.stream()
+				.filter(section -> section.isUpStation(target))
+				.findFirst()
+				.ifPresent(section -> section.updateUpStation(target));
+			this.sections.add(target);
+			return;
+		}
+
+		if (isExistDownStation(target)) {
+			this.sections.stream()
+				.filter(section -> section.isDownStation(target))
+				.findFirst()
+				.ifPresent(section -> section.updateDownStation(target));
+			this.sections.add(target);
+		}
+	}
+
+	public boolean isExistUpStation(Section section) {
```

Comment:
Sections 클래스 내부에서만 사용하니 private으로 만드는게 좋겠네요
그리고 아래와 같은 메서드를 만들면 isExistUpStation, isExistDownStation 같은 중복 로직을 제거할 수 있겠네요

```suggestion
	public boolean isExistStation(Station station) {
		return isContainStation(station);
	}
```